src/apps/btc/confirm_multisig.c

```diff
@@ -66,11 +66,29 @@ bool apps_btc_confirm_multisig(
     switch (coin) {
     case BTCCoin_BTC:
     case BTCCoin_LTC:
-        xpub_type = BTCPubRequest_XPubType_CAPITAL_ZPUB;
+        switch (multisig->script_type) {
+        case BTCScriptConfig_Multisig_ScriptType_P2WSH:
+            xpub_type = BTCPubRequest_XPubType_CAPITAL_ZPUB;
+            break;
+        case BTCScriptConfig_Multisig_ScriptType_P2WSH_P2SH:
+            xpub_type = BTCPubRequest_XPubType_CAPITAL_YPUB;
```

Comment:
I can feed the device a Ypub and a zpub like so but it will display both co-signers as Ypubs:

```python
        # somewhere in py/send_message.py
        account_keypath = [48 + HARDENED, 0 + HARDENED, 0 + HARDENED, 1 + HARDENED]
        my_xpub = self._device.btc_xpub(
            keypath=account_keypath,
            coin=bitbox02.btc.BTCCoin,
            xpub_type=bitbox02.btc.BTCPubRequest.CAPITAL_YPUB,
            display=False,
        )
        multisig_config = bitbox02.btc.BTCScriptConfig(
            multisig=bitbox02.btc.BTCScriptConfig.Multisig(
                threshold=1,
                xpubs=[
                    util.parse_xpub(my_xpub),
                    # random native segwit wallet generated with electrum.
                    util.parse_xpub(
                        "zpub6nKFyjXTxAqncB3QSCC4SVCahYBtVdf56CypKFs6V7gNhrgVJwa"
                        "1qSNKCpXFutShpsxtpa5uAB6S3wgtkMRucjiVFw38eVrqWBUSa57vWAH"
                    ),
                ],
                our_xpub_index=0,
                script_type=bitbox02.btc.BTCScriptConfig.Multisig.P2WSH_P2SH,
            )
        )

        # Accepts but displays a Ypub for xpubs[1] because util.parse_xpub above strips [:4]
        # and decodes the rest. 
        self._device.btc_register_script_config(
            coin=bitbox02.btc.BTCCoin,
            script_config=multisig_config,
            keypath=account_keypath,
            name="test",
        )

        # This also works.
        self._device.btc_address(
                    coin=bitbox02.btc.BTCCoin,
                    keypath=[48 + HARDENED, 0 + HARDENED, 0 + HARDENED, 1 + HARDENED, 0, 0],
                    script_config=multisig_config,
                    display=True,
        )

```

Is this intended? When I tried feeding this to Electrum, it refused.
I realize thought that the firmare has no way of knowing because it just receives the raw keys. Maybe it should receive encoded xpubs instead, in the future?

Reply:
same, `util.parse_xpub()` will strip the version bytes which make up the prefix. what goes over protobuf is the raw data without the version bytes.